### ISSUE TITLE
cmake takes about 15 minutes to go from -- Configuring done to -- Generating done

### ISSUE THREAD

**Issue by ChristophHaag** [GPUOpen-Drivers/xgl#17]
cmake version 3.10.2 on archlinux.

      cmake -H. -Bbuilds/Release64 -DCMAKE_BUILD_TYPE=Release -DCMAKE_MODULE_PATH=$PWD/../pal/cmake/Modules -DXGL_PAL_PATH:PATH=$PWD/../pal \
        -DCMAKE_C_FLAGS="-DLINUX -D__x86_64__ -D__AMD64__" -DCMAKE_CXX_FLAGS="-DLINUX -D__x86_64__ -D__AMD64__" -DXGL_LLVM_SRC_PATH=$PWD/../llvm-amdvlk

I'm not sure what's happening there but this seems excessive.

**Comment by davidmaoamd** [comment]
Could you paste me the log of cmake?
Thanks.

**Comment by ChristophHaag** [comment]
[CMakeOutput.log](https://github.com/GPUOpen-Drivers/xgl/files/1762552/CMakeOutput.log)
[CMakeError.log](https://github.com/GPUOpen-Drivers/xgl/files/1762553/CMakeError.log)

I have not experimented a lot with it yet. cmake is using 100% CPU on one core during this time and I just ran perf top and I think it's pretty clear
![screenshot_20180227_132138](https://user-images.githubusercontent.com/781076/36728510-4546bdd6-1bc1-11e8-9083-d6151e728f86.png)

**Comment by davidmaoamd** [comment]
Ever since last drop, the cmake command line has been largely simplified. 
Quote from README.md:
"cd <root of vulkandriver>/drivers/xgl
cmake -H. -Bbuilds/Release64
cd builds/Release64
make -j$(nproc)"
I noticed that you might have another llvm in place so you have to change the amdvlk llvm to the llvm-amdvlk. 
Could you try the new cmake command again with llvm-amdvlk renamed to llvm?

**Comment by ChristophHaag** [comment]
llvm is the amd-vulkan-master branch from https://github.com/GPUOpen-Drivers/llvm.git it was just put in a directory called llvm-amdvlk by the arch build script we use at https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=amdvlk-git

I changed it to the simpler cmake command (also tried without -DCMAKE_BUILD_TYPE=Release) but it still takes long.

If nobody else has a system where it reproduces I'll try cmake verbose/trace later to see where exactly it's taking so long.

**Comment by davidmaoamd** [comment]
I don't have the archlinux on hand.
And I cannot reproduce the issue after upgrading the cmake to 3.10.2 on my ubuntu box.
I will try to reproduce it first in the archlinux image by using the AUR later.
Thanks.

**Comment by davidmaoamd** [comment]
I tried archlinux today in the docker. 
I pull the image opamp/archlinux, and install the "base-devel cmake git" through pacman.
The cmake version is 3.10.2 and the compiler is gcc-7.3. 
I clone the project from https://aur.archlinux.org/amdvlk-git.git and run the "makepkg -s" under the amdvlk-git. 
Everything just works fine and the driver got built successfully. 
I then created another build directory under xgl and try to configure the cmake from scratch.
It just takes several seconds to complete.
Is there anything special configured in your system?
Did you get any user feedback on the similar issue?
Thanks.

**Comment by ChristophHaag** [comment]
The driver gets built successfully for me too, it just takes 15 minutes before it starts building.

It just showed this

```
-- Configuring done
CMake Error at /home/chris/build/amdvlk-git/src/pal/src/CMakeLists.txt:25 (add_library):
  Cannot find source file:

    /usr/include/qt/QtCompositor/5.5.0/QtCompositor/private/qwayland-server-wayland.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx
```
and I found some old broken symlinks there (no idea, I think it was from an old experiment and I cleaned up that old qt 5.5 build).

I think cmake may be searching through my entire /usr/include (trying to match something?) and so it may get slower the more headers you have installed.

I can't think of anything else that is special about my system. I have not heard anyone else complain about it but people might not be watching the build. llvm etc. is not quick to build so build times are expected to be not very quick.
